packages/protocol/contracts/governance/Quorum.sol

```diff
@@ -0,0 +1,160 @@
+pragma solidity ^0.5.8;
+
+import "fixidity/contracts/FixidityLib.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+import "./interfaces/IQuorum.sol";
+import "../common/Initializable.sol";
+
+
+/**
+ * @title Maintains a participation baseline and computes proposal support for Governance.
+ */
+contract Quorum is IQuorum, Ownable, Initializable {
```

Comment:
What are we at in terms of bytecode length for governance?

One thing that might make this code more readable would be to build and link a library for operating on proposals, which could include functions like `Proposal.isPassing`, `Proposal.isExpired`, `Proposal.getStage`, etc.

This would allow us to fold the quorum hyper-parameters into `Governance`.

Wdyt?

Reply:
Currently Governance is sitting at 23k bytes. With hotfixing expected to add some logic to the contract, keeping a healthy margin seems prudent.
While I agree that such a setup makes more sense and improves readability, without a suitable encoder for passing proposal structs into these functions, my understanding is that we wouldn't be able to call into this Proposal library without needing to perform some potentially messy unpacking. Do you have a workaround in mind for this?

Reply:
It seems like an analogous situation to the LinkedList libraries, which operate on structs and are linked. Any reason we wouldn't be able to do the same thing here?

Reply:
Oh good point. I guess passing in storage structs is okay